knx/dpt/types_registry.go

```diff
@@ -81,36 +75,30 @@ func (r *Registry) Init() error {
 
 	// Register the types
 	for _, d := range dpts {
-		err := r.add(d)
+		err = r.add(d)
 		if err != nil {
-			return err
+			return
 		}
 	}
 
-	return nil
+	return
 }
 
 func (r Registry) List() []string {
 	// Initialize the key-list
-	keys := make([]string, len(r.lut))
+	keys := make([]string, len(r))
 
 	// Fill the key-list
 	i := 0
-	for k := range r.lut {
+	for k := range r {
 		keys[i] = k
 		i++
 	}
 
 	return keys
 }
 
-func (r Registry) Lookup(name string) (t DatapointValue, err error) {
-	t, ok := r.lut[name]
-	if ok {
-		err = nil
-	} else {
-		err = fmt.Errorf("datapoint-type \"%v\" not found!", name)
-	}
-
+func (r Registry) Lookup(name string) (d DatapointValue, ok bool) {
+	d, ok = r[name]
```

Comment:
What happens if any of these `DatapointValue`s is reused? For example:

```go
a = registry["1.001"]
b = registry["1.001"]

a.Unpack(data1)
b.Unpack(data2)
```

Won't `a` and `b` be the same therefore the second call to `Unpack` would override the previous unpacking?

Reply:
You are absolutely right and I was not thinking on this as my current use is to copy-away the value in between the two unpack calls... But it's not nice indeed!